examples/cms-starter/src/FieldMapping.tsx

```diff
@@ -0,0 +1,233 @@
+                if (field?.field?.type === "string") {
+                    const currentSlugField = fields.find(field => field.field?.id === slugFieldId)
+                    if (!currentSlugField || nextSet.has(slugFieldId ?? "")) {
+                        setSlugFieldId(fieldId)
+                    }
+                }
+            } else {
+                nextSet.add(fieldId)
+
+                // If the disabled field is the slug field, update it to the next
+                // possible slug field
+                if (fieldId === slugFieldId) {
+                    const nextSlugField = possibleSlugFields.find(field => field.id !== fieldId)
+                    if (nextSlugField) {
+                        setSlugFieldId(nextSlugField.id)
+                    }
+                }
+            }
+            return nextSet
+        })
+    }
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        if (!slugFieldId) {
+            framer.notify("Slug field is required", {
+                variant: "error",
+            })
+            return
+        }
+
+        try {
+            setIsSyncing(true)
+            await onSubmit(
+                dataSource,
+                fields.filter(field => field.field && !disabledFieldIds.has(field.field!.id)),
+                slugFieldId
+            )
+            await framer.closePlugin(`Synchronization successful`, {
+                variant: "success",
+            })
+        } catch (error) {
+            console.error(error)
+            framer.notify(`Failed to sync collection ${dataSource.id}`, {
+                variant: "error",
+            })
+        } finally {
+            setIsSyncing(false)
+        }
+    }
+
+    useLayoutEffect(() => {
+        framer.showUI({
+            width: 360,
+            height: 425,
+            minWidth: 360,
+            minHeight: 425,
+            resizable: true,
+        })
+    }, [])
+
+    return (
+        <main className="mapping no-scrollbar">
+            <form onSubmit={handleSubmit}>
+                <hr className="sticky-top" />
+                <label className="slug-field" htmlFor="slugField">
+                    Slug Field
+                    <select
+                        name="slugField"
+                        className="field-input"
+                        value={slugFieldId ?? ""}
+                        onChange={e => setSlugFieldId(e.target.value)}
+                        required
+                    >
+                        {possibleSlugFields.map(field => {
+                            if (!field) return null
+                            return (
+                                <option key={field.id} value={field.id}>
+                                    {field.name}
+                                </option>
+                            )
+                        })}
```

Comment:
I can't select any fields for either of the two collections

Reply:
Fixed!